app/assets/javascripts/application.js

```diff
@@ -1,3 +1,30 @@
 //= require jquery
 //= require jquery_ujs
 //= require cfa_styleguide_main
+
+var toggleDisabledField = (function() {
+    var f = {
+        init: function() {
+            var $checkbox = $('.checkbox-with-associated-field-to-disable input'),
+                $fieldToDisable = $('.associated-field-to-disable');
+
+            toggle = function($checkbox, $fieldToDisable) {
+                $fieldToDisable.find('input').prop('disabled', $checkbox.is(':checked'));
+                $fieldToDisable.toggleClass('disabled', $checkbox.is(':checked') );
+            };
+
+            toggle($checkbox, $fieldToDisable);
+
+            $checkbox.change(function (e) {
+                toggle($checkbox, $fieldToDisable);
+            });
```

Comment:
I think this javascript is fine. It has a couple of issues that I don't think are too important given the timeline. they are:

- the `toggle` variable doesn't have a `var` declaration
- I think this will only work on a single element. It probably would be better to have a `data-disable-target` or something so that multiple toggles/disableable-elements could coexist, but we can cross that bridge later.